source/agora/consensus/protocol/Nominator.d

```diff
@@ -179,6 +179,13 @@ extern(D):
         if (data.tx_set.length == 0)
             return;  // not ready yet
 
+        // check whether the consensus data is valid before nominating it.
+        if (auto msg = this.ledger.validateConsensusData(data))
+        {
+            log.fatal("tryNominate(): Invalid consensus data: {}", msg);
```

Comment:
It's not fatal if the node continues running. Fatal means the node will shut down.

Reply:
Ok it looks like @Geod24 wants this to be more than `log.error`, but we don't have `log.critical` for this purpose. So `log.fatal` is the next best thing, despite its name.

Keep this changeset.

Reply:
Do we want to log out the actual consensus data too and not just the message?

Reply:
> Do we want to log out the actual consensus data too and not just the message?

Logging out the data would be good.